purchases/src/defaults/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -223,142 +221,6 @@ class Purchases internal constructor(
         purchasesOrchestrator.purchase(purchaseParams, callback)
     }
 
-    /**
-     * Purchases [storeProduct].
-     * If [storeProduct] represents a subscription, upgrades from the subscription specified by
-     * [upgradeInfo.oldSku] and chooses [storeProduct]'s default [SubscriptionOption].
-     *
-     * The default [SubscriptionOption] logic:
-     *   - Filters out offers with "rc-ignore-offer" tag
-     *   - Uses [SubscriptionOption] WITH longest free trial or cheapest first phase
-     *   - Falls back to use base plan
-     *
-     * If [storeProduct] represents a non-subscription, [upgradeInfo] will be ignored.
-     *
-     * @param [activity] Current activity
-     * @param [storeProduct] The StoreProduct of the product you wish to purchase
-     * @param [upgradeInfo] The upgradeInfo you wish to upgrade from, containing the oldSku and the optional
-     * prorationMode. Amazon Appstore doesn't support changing products so upgradeInfo is ignored for Amazon purchases.
-     * @param [listener] The PurchaseCallback that will be called when purchase completes.
-     */
-    @Deprecated(
-        "Use purchase() and PurchaseParams.Builder instead",
-        ReplaceWith("purchase()"),
-    )
-    fun purchaseProduct(
-        activity: Activity,
-        storeProduct: StoreProduct,
-        upgradeInfo: UpgradeInfo,
-        listener: ProductChangeCallback,
-    ) {
-        purchasesOrchestrator.startDeprecatedProductChange(
-            activity,
-            storeProduct.purchasingData,
-            null,
-            upgradeInfo.oldSku,
-            GoogleReplacementMode.fromPlayBillingClientMode(upgradeInfo.replacementMode),
-            listener,
-        )
-    }
-
-    /**
-     * Purchases a [StoreProduct]. If purchasing a subscription, it will choose the default [SubscriptionOption].
-     *
-     * The default [SubscriptionOption] logic:
-     *   - Filters out offers with "rc-ignore-offer" tag
-     *   - Uses [SubscriptionOption] WITH longest free trial or cheapest first phase
-     *   - Falls back to use base plan
-     *
-     * @param [activity] Current activity
-     * @param [storeProduct] The StoreProduct of the product you wish to purchase
-     * @param [callback] The PurchaseCallback that will be called when purchase completes.
-     */
-    @Deprecated(
-        "Use purchase() and PurchaseParams.Builder instead",
-        ReplaceWith("purchase()"),
-    )
-    fun purchaseProduct(
```

Comment:
IMO I would leave these functions that don't have a `UpgradeInfo`. I think some people might still be using those, and we should try to minimize the changes if possible. Lmk if you think otherwise though!

Reply:
Haha I'm fine with both, but I went off of @vegaro's [comment](https://github.com/RevenueCat/purchases-android/pull/1707#discussion_r1618415370): 
> only leave the `purchase` function that takes `PurchaseParams.Builder` 
  
I'm happy to just do what the 2 of you agree on 😄

Reply:
These were deprecated in the current major https://github.com/RevenueCat/purchases-android/pull/825, that's where my suggestion for removal came from. If we don't remove them now, they will live for another major. Current major was released in September, I think 9 months is enough time to migrate to the `purchase` function?

Reply:
Right, but I think it shouldn't be a big problem to keep those around right? And I think as long as it's not a problem, we should keep the deprecated functions. If at some point it becomes a problem, we can reconsider.

Reply:
Agreed, I added it to our sync. Let's discuss it then!

Reply:
I spoke to @tonidero yesterday and have concluded that it is probably better to keep them, so we focus on the developers and not on our own development experience. We can always remove them in the future if they become a pain to maintain. I think it's still worth to discuss in the meeting today so we all align for the future though.

Sorry for the back and forth, and for making you work on something that is getting scratched 😢

Reply:
No worries, it's good exercise! 😄 I'll await today's meeting just in case, and will make the changes afterwards.

Reply:
This is now done!